shared/common-adapters/header-hoc.types.js.flow

```diff
@@ -1,15 +1,32 @@
 // @flow
 import * as React from 'react'
 import {type StylesCrossPlatform} from '../styles'
+import {type IconType} from './icon.constants'
+export type Action = {
+  custom?: React.Node,
+  label?: string, // TODO: make this required after updates are fully integrated
```

Comment:
can't make this required since it's not needed for custom / icon, right? Rather than have a bunch of optional fields with implicit, I think this might be clearer:

```
type Action = {|
  type: 'custom',
  node: React.Node,
|} | {|
  type: 'icon',
  icon: IconType,
  onPress?: Function,
|} | {|
  type: 'text',
  label: string,
  onPress?: Function,
|}
```

and then you can switch on `type` in `render()`.

Reply:
I was requiring the label for use when there are more than 3 actions (it's the string that shows up on the menu item). I also wanted to make it easier to create actions by not having to specify a `type`.